dll/ntdll/rtl/hashtable.c

```diff
@@ -0,0 +1,1264 @@
+// Placeholder for RTL_DYNAMIC_HASH_TABLE structure and related functions
+// For tests, create a separate file per each function under
+// .\reactos\modules\rostests\apitests\ntdll
+// TODO: Write tests per public function
+
+/*
+ * COPYRIGHT:       See COPYING in the top level directory
+ * PROJECT:         ReactOS NT User-Mode DLL
+ * FILE:            lib/ntdll/rtl/hashtable.c
+ * PURPOSE:         Hashtable implementation in kernel
+ * PROGRAMMERS:     Zafer Balkan <zafer@zaferbalkan.com>
```

Comment:
```suggestion
/*
 * PROJECT:     ReactOS NT User-Mode DLL
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
 * PURPOSE:     Hashtable implementation in kernel
 * COPYRIGHT:   Copyright 2023 Zafer Balkan <zafer@zaferbalkan.com>
```
The header is slightly malformatted, I assume that you took a reference on outdated existing code.
Here is the current template (we do especially not want to have the filename referenced in files any longer):
https://reactos.org/wiki/Coding_Style#File_Structure